bot/exts/utils/reminders.py

```diff
@@ -30,6 +32,133 @@
+            amount, unit = selected_duration.split()
+            if unit.startswith("month"):
+                num_seconds = (current_datetime + relativedelta(months=int(amount)) - current_datetime).total_seconds()
+
+            else:  # unit is years
+                num_seconds = (current_datetime + relativedelta(years=int(amount)) - current_datetime).total_seconds()
+
+        new_reminder_end_time = current_datetime + relativedelta(seconds=num_seconds)
+
+        await self.bot.api_client.patch(
+            f'bot/reminders/{self.reminder_id}',
+            json={'expiration': new_reminder_end_time.isoformat()}
+        )
+
+        self.new_expiry = new_reminder_end_time
+        new_end_timestamp = time.discord_timestamp(new_reminder_end_time, format=time.TimestampFormats.DAY_TIME)
+        original_embed = interaction.message.embeds[0]
+        new_embed = discord.Embed(
+            colour=discord.Colour.green(),
+            title=random.choice(POSITIVE_REPLIES),
+            description=f"Successfully snoozed reminder for {selected_duration}. New remind time: {new_end_timestamp}",
+        )
+        new_embed.set_footer(text=f"ID: {self.reminder_id}")
+        try:
+            await interaction.response.edit_message(embeds=[original_embed, new_embed], view=None)
+        except discord.NotFound:
+            await interaction.message.channel.send(embed=new_embed)
+
+        self.reminder_was_snoozed = True
+        self.stop()
+
+
+class SnoozeButtonView(discord.ui.View):
+    """The reminder's snooze button UI View."""
+
+    def __init__(self, bot: Bot, reminder_id: int, reminder_author_id: int):
+        super().__init__(timeout=300)
+        self.new_expiry: t.Optional[datetime] = None
+        self.reminder_was_snoozed = False
+
+        self.bot = bot
+        self.reminder_id = reminder_id
+        self.reminder_author_id = reminder_author_id
+
+        snooze_button = SnoozeButton(self)
+        self.add_item(snooze_button)
+
+
+class SnoozeButton(discord.ui.Button):
+    """The reminder's snooze button."""
+
+    def __init__(self, parent_view: SnoozeButtonView):
+        super().__init__(label="Snooze Reminder")
+        self.parent_view = parent_view
+
+    async def callback(self, interaction: discord.Interaction) -> None:
+        """Ensure the reminder author matches the user who clicked button and then display select view."""
+        if interaction.user.id != self.parent_view.reminder_author_id:
+            await interaction.response.send_message(":x: This is not your reminder to snooze!", ephemeral=True)
+            return
+
+        snooze_select_view = SnoozeSelectView(
+            bot=self.parent_view.bot,
+            reminder_id=self.parent_view.reminder_id,
+            reminder_author_id=self.parent_view.reminder_author_id
+        )
+        await interaction.response.edit_message(view=snooze_select_view)
+
+        await snooze_select_view.wait()
+
+        if snooze_select_view.reminder_was_snoozed:
```

Comment:
You can use the return value of `wait()` instead! It returns a boolean which indicates whether the view timed out or was stopped (I don't remember which case had which boolean value though).

Reply:
Yep! Have updated :+1: